### PR TITLE
NIFI-9104-9113 Multiple JUnit 5 refactorings (Druid -> GRPC)

### PR DESCRIPTION
**Pull request by MikeThomsen**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/test/java/org/apache/nifi/processors/email/TestListenSMTP.java [hunk 1]
```diff
@@ -44,7 +28,23 @@
 import org.apache.nifi.util.TestRunner;
 import org.apache.nifi.util.TestRunners;
 import org.junit.BeforeClass;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/test/java/org/apache/nifi/processors/email/TestListenSMTP.java:30]: This class is still referencing to JUnit 4 `BeforeClass` annotation, as opposed to the JUnit 5 `BeforeAll` annotation.  This is causing the test to fail with a NullPointerException due to the missing TLS configuration.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]

File: nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/ITListenGRPC.java [hunk 2]
```diff
@@ -19,26 +19,57 @@
...
+    }
+
+    private static void useSSLContextService(final TestRunner controller, final Map<String, String> sslProperties) {
+        final SSLContextService service = new StandardSSLContextService();
+        assertDoesNotThrow(() -> controller.addControllerService("ssl-service", service, sslProperties));
+        controller.enableControllerService(service);
+
+        controller.setProperty(InvokeGRPC.PROP_SSL_CONTEXT_SERVICE, "ssl-service");
+    }
+
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/ITListenGRPC.java:72]: Are these changes necessary?

File: nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/ITListenGRPC.java [hunk 3]
```diff
@@ -119,17 +151,67 @@ public void testOutOfSpaceRoundTrip() throws Exception {
...
         final FlowFileServiceGrpc.FlowFileServiceBlockingStub stub = FlowFileServiceGrpc.newBlockingStub(channel);
 
         final ListenGRPC listenGRPC = new ListenGRPC();
         final TestRunner runner = TestRunners.newTestRunner(listenGRPC);
+        runner.setAllowSynchronousSessionCommits(true);
         runner.setProperty(ListenGRPC.PROP_SERVICE_PORT, String.valueOf(randPort));
-        // set max message size to 1 byte to force exception to be thrown.
-        runner.setProperty(ListenGRPC.PROP_MAX_MESSAGE_SIZE, "1B");
+        runner.setProperty(ListenGRPC.PROP_USE_SECURE, "true");
+        useSSLContextService(runner, getKeystoreProperties());
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/ITListenGRPC.java:214]: Is this change necessary?

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Most of the changes look good, thanks @MikeThomsen!

I noticed a number of additions in `ITListenGRPC`, it is possible that those additions were the result of a rebase that introduced TLS testing where it was previously removed?  Pulling the current revision and making more selective adjustments should resolve the issue.  Other than that one file, this should be good to go.

File: nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/ITListenGRPC.java (hunk 2 above)
Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/ITListenGRPC.java:72]: Might be a rebase. I'll copy pasta the current contents in and then fix those up since it feels like some rebase ugliness.

**Review by exceptionfactory — APPROVED** [review]
Thanks for updating the GRPC integration test @MikeThomsen, the changes look good! +1 merging.
